core/templates/hash_map.h

```diff
@@ -588,13 +600,22 @@ class HashMap {
 		}
 	}
 
-	HashMap(uint32_t p_initial_capacity) {
-		// Capacity can't be 0.
-		capacity_index = 0;
+	HashMap(uint32_t p_initial_capacity) :
+			HashMap() {
 		reserve(p_initial_capacity);
 	}
 	HashMap() {
 		capacity_index = MIN_CAPACITY_INDEX;
+		uint32_t capacity = hash_table_size_primes[capacity_index];
+		// Allocate on demand to save memory.
+
+		hashes = reinterpret_cast<uint32_t *>(Memory::alloc_static(sizeof(uint32_t) * capacity));
+		elements = reinterpret_cast<HashMapElement<TKey, TValue> **>(Memory::alloc_static(sizeof(HashMapElement<TKey, TValue> *) * capacity));
+
+		for (uint32_t i = 0; i < capacity; i++) {
+			hashes[i] = EMPTY_HASH;
+			elements[i] = nullptr;
+		}
```

Comment:
Why not use `reserve` here? (And remove the constructor call to this one in the other two constructors)

Reply:
> Why not use the reserve here?

The compiler optimizes so that the loop has the same performance as memcpy.

Reply:
> And remove the constructor call to this one in the other two constructors

It is necessary because reserve will not work without it. 
``` c++
if (elements == nullptr) {
	capacity_index = new_index;
	return; // Unallocated yet.
}
```
Even if you remove the check, the unallocated memory will be deleted, which will cause an error.